packages/console/src/ConsoleStatusBar.tsx

```diff
@@ -1,136 +1,109 @@
-}
-
-interface ConsoleStatusBarState {
-  isDisconnected: boolean;
-  isCommandRunning: boolean;
+  overflowActions?: DropdownActions;
 }
 
-export class ConsoleStatusBar extends PureComponent<
-  ConsoleStatusBarProps,
-  ConsoleStatusBarState
-> {
-  static defaultProps = {
-    children: null,
-  };
-
-  constructor(props: ConsoleStatusBarProps) {
-    super(props);
-
-    this.handleCommandStarted = this.handleCommandStarted.bind(this);
-    this.handleCommandCompleted = this.handleCommandCompleted.bind(this);
-
-    this.pending = new Pending();
-
-    this.state = {
-      isDisconnected: false,
-      isCommandRunning: false,
-    };
-  }
-
-  componentDidMount(): void {
-    this.startListening();
-  }
-
-  componentWillUnmount(): void {
-    this.stopListening();
-    this.cancelPendingPromises();
-  }
-
-  pending: Pending;
-
-  startListening(): void {
-    const { dh, session } = this.props;
-    session.addEventListener(
-      dh.IdeSession.EVENT_COMMANDSTARTED,
-      this.handleCommandStarted
-    );
-  }
-
-  stopListening(): void {
-    const { dh, session } = this.props;
-    session.removeEventListener(
-      dh.IdeSession.EVENT_COMMANDSTARTED,
-      this.handleCommandStarted
-    );
-  }
-
-  cancelPendingPromises(): void {
-    this.pending.cancel();
-  }
-
-  handleCommandStarted(event: CustomEvent): void {
-    const { result } = event.detail;
-
-    this.pending
-      .add(result)
-      .then(() => this.handleCommandCompleted(null))
-      .catch(error => this.handleCommandCompleted(error));
-
-    this.setState({
-      isCommandRunning: true,
-    });
-  }
-
-  handleCommandCompleted(error: unknown): void {
-    // Don't update state if the promise was canceled
-    if (!(error instanceof CanceledPromiseError)) {
-      this.setState({
-        isCommandRunning: this.pending.pending.length > 0,
-      });
+function ConsoleStatusBar({
+  children,
+  dh,
+  session,
+  overflowActions,
+}: ConsoleStatusBarProps): ReactElement {
+  const [pendingCommandCount, setPendingCommandCount] = useState(0);
+
+  const handleCommandStarted = useCallback(async (event: CustomEvent) => {
+    setPendingCommandCount(count => count + 1);
+
+    try {
+      const { result } = event.detail;
+      await result;
+    } catch (error) {
+      // No-op, fall through
     }
-  }
-
-  render(): ReactElement {
-    const { children, dh, openObject, overflowActions, objects } = this.props;
-    const { isDisconnected, isCommandRunning } = this.state;
 
-    let statusIconClass = null;
-    let tooltipText = null;
+    setPendingCommandCount(count => count - 1);
+  }, []);
+
+  useEffect(
+    function startListening() {
+      session.addEventListener(
+        dh.IdeSession.EVENT_COMMANDSTARTED,
+        handleCommandStarted
+      );
+
+      return function stopListening() {
+        session.removeEventListener(
+          dh.IdeSession.EVENT_COMMANDSTARTED,
+          handleCommandStarted
+        );
+      };
```

Comment:
Looks like `session.addEventListener` returns a cleanup function we could use instead